internal/controller/nonadminbackup_controller.go

```diff
@@ -161,28 +161,19 @@ func (r *NonAdminBackupReconciler) ValidateSpec(ctx context.Context, logrLogger
 	// Main Validation point for the VeleroBackup included in NonAdminBackup spec
 	_, err := function.GetBackupSpecFromNonAdminBackup(nab)
 	if err != nil {
-		updated := updateNonAdminPhase(&nab.Status.Phase, nacv1alpha1.NonAdminBackupPhaseBackingOff)
-		if updated {
-			if updateErr := r.Status().Update(ctx, nab); updateErr != nil {
-				logger.Error(updateErr, phaseUpdateError)
-				return true, false, updateErr
-			}
-
-			logger.V(1).Info(phaseUpdateRequeue)
-			return false, true, nil
-		}
-
-		updated = meta.SetStatusCondition(&nab.Status.Conditions,
+		updatedPhase := updateNonAdminPhase(&nab.Status.Phase, nacv1alpha1.NonAdminBackupPhaseBackingOff)
+		updatedCondition := meta.SetStatusCondition(&nab.Status.Conditions,
 			metav1.Condition{
 				Type:    string(nacv1alpha1.NonAdminConditionAccepted),
 				Status:  metav1.ConditionFalse,
 				Reason:  "InvalidBackupSpec",
 				Message: err.Error(),
 			},
 		)
-		if updated {
+		if updatedPhase || updatedCondition {
 			if updateErr := r.Status().Update(ctx, nab); updateErr != nil {
-				logger.Error(updateErr, conditionUpdateError)
+				// TODO error message
```

Comment:
should I change error message here, now can be phase and conditions update error

Better to be more generic? Like `Failed to update NonAdminBackup Status`?

Reply:
yes generic IMO

Reply:
maybe append some task related context to the generic message prefix

Reply:
will update message